samples/CommunityToolkit.Maui.Markup.Sample/AppShell.cs

```diff
@@ -0,0 +1,61 @@
+﻿using CommunityToolkit.Maui.Markup.Sample.Pages;
+using CommunityToolkit.Maui.Markup.Sample.Pages.Base;
+using CommunityToolkit.Maui.Markup.Sample.ViewModels;
+using CommunityToolkit.Maui.Markup.Sample.ViewModels.Base;
+
+namespace CommunityToolkit.Maui.Markup.Sample;
+
+class AppShell : Shell
+{
+	readonly static IReadOnlyDictionary<Type, string> pageRouteMappingDictionary = new Dictionary<Type, string>(new[]
+	{
+		CreateRoutePageMapping<NewsPage, NewsViewModel>(),
+		CreateRoutePageMapping<SettingsPage, SettingsViewModel>(),
+		CreateRoutePageMapping<NewsDetailPage, NewsDetailViewModel>(),
+	});
+
+	public AppShell(NewsPage newsPage)
+	{
+		Items.Add(newsPage);
+	}
+
+	public static string GetRoute<TPage, TViewModel>() where TPage : BaseContentPage<TViewModel>
+														where TViewModel : BaseViewModel
+	{
+		if (!pageRouteMappingDictionary.ContainsKey(typeof(TPage)))
```

Comment:
Can we use the `TryGet` method? That way we will do the _Key verification_ and grab the value in one move.

Reply:
Great idea! Done ✅